disnake/guild.py

```diff
@@ -2224,14 +2232,26 @@ async def create_scheduled_event(
             The request failed.
         TypeError
             The ``image`` asset is a lottie sticker (see :func:`Sticker.read`),
-            or one of ``entity_type``, ``privacy_level``, or ``entity_metadata``
-            is not of the correct type.
+            one of ``entity_type``, ``privacy_level``, or ``entity_metadata``
+            is not of the correct type, or the ``entity_type`` was not provided and
+            could not be assumed from the ``channel``.
 
         Returns
         -------
         :class:`GuildScheduledEvent`
             The newly created guild scheduled event.
         """
+
+        if channel is None:
+            entity_type = GuildScheduledEventEntityType.external
+        elif hasattr(channel, "type") and isinstance(channel.type, ChannelType):  # type: ignore
+            if channel.type == ChannelType.voice:  # type: ignore
+                entity_type = GuildScheduledEventEntityType.voice
+            elif channel.type == ChannelType.stage_voice:  # type: ignore
+                entity_type = GuildScheduledEventEntityType.stage_instance
+        elif entity_type is MISSING:
+            raise TypeError("entity_type must be provided if the channel does not have a type")
```

Comment:
This logic has a bit of an issue where the channel has a type that isn't stage_voice or voice. 

We should also raise TypeError if the type isn't voice or stage_voice.

Reply:
fixed in https://github.com/DisnakeDev/disnake/pull/617/commits/119f3081fc04dbe12c52401d83cf8042f4c12eb5